File: app/routes/google_drive.py
import os
from fastapi import APIRouter

from app.platforms.google_drive.auth import authenticate

TOKEN_FILE = "credentials/token.json"
router = APIRouter(
    prefix="/platforms/google-drive",
    tags=["Google Drive"]
)


@router.get("/connect")
def connect_google_drive():

    if os.path.exists(TOKEN_FILE):

        return {
            "status": "success",
            "connected": True,
            "message": "Google Drive already connected."
        }

    try:

        authenticate()

        return {
            "status": "success",
            "connected": True,
            "message": "Google Drive connected successfully."
        }

    except Exception as e:

        return {
            "status": "error",
            "connected": False,
            "message": str(e)
        }

@router.get("/status")
def google_drive_status():

    return {
        "connected": os.path.exists(TOKEN_FILE)
    }

@router.post("/disconnect")
def disconnect_google_drive():

    try:

        if os.path.exists(TOKEN_FILE):
            os.remove(TOKEN_FILE)

        return {
            "status": "success",
            "connected": False,
            "message": "Google Drive disconnected successfully."
        }

    except Exception as e:

        return {
            "status": "error",
            "connected": True,
            "message": str(e)
        }

File: frontend/src/services/googleDrive.ts
import API_BASE_URL from "./api";

export async function connectGoogleDrive() {

    const response = await fetch(
        `${API_BASE_URL}/platforms/google-drive/connect`
    );

    const data = await response.json();

    if (!response.ok) {
        throw new Error(data.message || "Unable to connect Google Drive.");
    }

    return data;
}

export async function getGoogleDriveStatus() {

    const response = await fetch(
        `${API_BASE_URL}/platforms/google-drive/status`
    );

    const data = await response.json();

    if (!response.ok) {
        throw new Error("Unable to get Google Drive status.");
    }

    return data;
}

export async function disconnectGoogleDrive() {

    const response = await fetch(
        `${API_BASE_URL}/platforms/google-drive/disconnect`,
        {
            method: "POST"
        }
    );

    const data = await response.json();

    if (!response.ok) {
        throw new Error(data.message || "Unable to disconnect Google Drive.");
    }

    return data;
}